src/cockpit/channels/filesystem.py

```diff
@@ -118,15 +117,28 @@ class FsReplaceChannel(Channel):
     _path = None
     _tag = None
     _tempfile = None
+    _temppath = None
 
     def do_open(self, options):
         self._path = options.get('path')
         self._tag = options.get('tag')
 
     def do_data(self, data):
         if self._tempfile is None:
-            dirname, basename = os.path.split(self._path)
-            self._tempfile = tempfile.NamedTemporaryFile(dir=dirname, prefix=f'.{basename}-', delete=False)
+            for i in range(1000):
```

Comment:
I'm not a fan either. It's what the C bridge does, and I discussed the other ideas that I had above. Other ideas gladly appreciated!

WDYM with _RandomNameSequence? That'd be something like

    for i in random.Random().choices(range(1000), k=1000):

which is hardly simpler, nor any better?

Reply:
Another option would be to not have a loop at all and just append `.cockpit-tmp`. For two concurrent replace()s that would mean that the first one wins instead of the last one; but that would also feel slightly wrong, and if you ever get a bridge crash right there, nothing would ever clean up the .cockpit.tmp file again. This could only work if we let go of the `O_EXCL` and accept overwriting a stale file. (But then, symlink attacks!)

Curiously we even have an unit test for that ("modify with conflict") which fails with just a single static name.

Reply:
No, sorry, I'm not going to do that. This is very far from "ideal" wrt. readability and testability.

Reply:
OK, changed to a random 6-character string.